### PR TITLE
Attempt to move GDT's integration test to the allowed_failures section

### LINKED ISSUE #2982: Flaky GDT end-to-end test
The flakiness of -[GDTIntegrationTest testEndToEndEvent] frequently causes Travis failures.

See https://travis-ci.org/firebase/firebase-ios-sdk/jobs/530603139 for an example.

### PR DESCRIPTION
**Pull request by mikehaney24**
Hopefully in the future it will become more obvious what the real problem is.

Fixes #2982

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -140,6 +140,15 @@ jobs:
         - travis_retry ./scripts/if_changed.sh ./scripts/build.sh $PROJECT $PLATFORM
         - travis_retry ./scripts/if_changed.sh ./scripts/pod_lib_lint.sh GoogleDataTransport.podspec
 
+    # GoogleDataTransport integration tests using the default Xcode version.
```
Comment (paulb777) [inline .travis.yml]: Why is the same block added here and below?

Comment (mikehaney24) [reply .travis.yml]: AFAICT, I had to factor out the integration test run into its own travis 'project' so I could tell travis to ignore it if it failed. If I added it only into the allow_failures section, travis wouldn't run it at all. All the projects listed in that section are written up here, except for the Firestore fuzz test which runs on a cronjob.

Could I write the allow_failures rule like so?:

```
  - stage: test
      env:
        - PROJECT=GoogleDataTransportIntegrationTest PLATFORM=iOS METHOD=xcodebuild
```

File: .travis.yml [hunk 2]
```diff
@@ -329,6 +338,14 @@ jobs:
   # before reporting the results.
   # fast_finish: true
 
+    - stage: test
```
Comment (paulb777) [inline .travis.yml]: It looks like it should only be the -env setting in the allow_failures section

File: .travis.yml (hunk 1 above)
Comment (paulb777) [reply .travis.yml]: see below

**Pull request merged**